src/s3ql/backends/gdrive.py

```diff
@@ -0,0 +1,599 @@
+                return True
+            #Google best practice error says error 500 could be temporal, also We get some resumable errors with 400 Bad Request (Fully Random)
+            elif exc.resp.status >= 500 or exc.resp.status == 400:
+                return True
+        elif isinstance(exc,(BrokenPipeError,ConnectionResetError,socket.gaierror,socket.herror,socket.timeout,http.client.IncompleteRead,
+        httplib2.ServerNotFoundError,ssl.SSLZeroReturnError, ssl.SSLEOFError,ssl.SSLSyscallError,ssl.SSLError)):
+            return True
+        return False
+
+    @staticmethod
+    def _escape_string(name):
+        '''Escape string'''
+        return name.replace("\\", "\\\\").replace("'", "\\'")
+
+    def _list_files(self, folder, fields, query=None):
+        '''List folder contents
+        Arguments:
+        folder -- Google Drive folder to list files in
+        fields -- file properties to return
+        query -- files.list query (optional)
+        '''
+        # build query
+        if query is None:
+            query = ''
+        else:
+            query += ' and '
+        query += "'{0}' in parents".format(folder['id'])
+        query += " and trashed = false"
+        request = self.service.files().list(q=query,pageSize=1000,fields="nextPageToken, files(%s)" % fields)
+        while request is not None:
+            results = request.execute()            
+            yield from results.get('files', [])
+            request = self.service.files().list_next(request,results)
+
+    def _lookup_file(self, name, folder=None):
+        '''Lookup file by name
+        Arguments:
+        name -- file name. May contain absolute or relative file name with '/' as path separator
+        folder -- folder to start lookup from (root folder if omitted)
+        '''
+        if folder is None:
+            folder = { 'id': 'root', 'path': '/', 'mimeType': Backend.MIME_TYPE_FOLDER }
+
+        # find next path component
+        if len(name) == 0:
+            return folder
+        if name[0] == '/':
+            name = name[1:]
+        n = name.split('/', 1)
+        if len(n[0]) == 0:
+            return folder
+
+        # iterate over folder children
+        query = "name = '{0}'".format(self._escape_string(n[0]))
+        for f in self._list_files(folder, "id, name, mimeType, size,md5Checksum, properties", query):
+            f['path'] = folder['path'] + f['name']
+            if f['mimeType'] == Backend.MIME_TYPE_FOLDER:
+                f['path'] += '/'
+
+            log.debug("{0} ({1}: {2})".format(f['path'], f['id'], f['mimeType']))
+            if len(n) == 1:
+                return f
+            elif f['mimeType'] == Backend.MIME_TYPE_FOLDER:
+                return self._lookup_file(n[1], f)
+            else:
+                return None
+
+        # nothing found
+        return None
+
+    @staticmethod
+    def _chunk_property(k, i):
+        '''Make chunk property'''
+        return 'ref({0}.{1})'.format(k, i)
```

Comment:
This really doesn't justify a separate method

Reply:
to be refactored